tests/adapters_t/batchsystems_t/test_slurm.py

```diff
@@ -0,0 +1,242 @@
+        #  self.command_wo_options = 'sinfo --Format="statelong,cpusstate,allocmem,memory,features,nodehost" -e --noheader'
+
+        return_value = "\n".join(
+            [
+                f"mixed      2/2/0/4   6000    24000   VM-1   host-10-18-1-1",
+                f"mixed      3/1/0/4   15853   22011   VM-2   host-10-18-1-2",
+                f"mixed      1/3/0/4   18268   22011   VM-3   host-10-18-1-4",
+                f"mixed      3/1/0/4   17803   22011   VM-4   host-10-18-1-7",
+                f"draining   0/4/0/4   17803   22011   draining_machine   draining_machine",
+                f"idle       0/4/0/4   17803   22011   idle_machine   idle_machine",
+                f"drained    0/4/0/4   17803   22011   drained_machine   drained_machine",
+                f"powerup    0/4/0/4   17803   22011   power_up_machine   power_up_machine",
+            ]
+        )
+
+        self.mock_async_run_command.return_value = async_return(
+            return_value=return_value
+        )
+
+        self.setup_config_mock(options={"partition": "test_part"})
+
+        self.slurm_adapter = SlurmAdapter()
+
+    def tearDown(self):
+        self.mock_async_run_command.reset_mock()
+
+    def setup_config_mock(self, options=None):
+        self.config = self.mock_config.return_value
+        self.config.BatchSystem.ratios = {
+            "cpu_ratio": "Real(TotalSlotCpus-Cpus)/TotalSlotCpus",
+            "memory_ratio": "Real(TotalSlotMemory-Memory)/TotalSlotMemory",
+        }
+        self.config.BatchSystem.max_age = 10
+        if options:
+            self.config.BatchSystem.options = options
+        else:
+            self.config.BatchSystem.options = {}
+
+    def test_disintegrate_machine(self):
+        self.assertIsNone(
+            run_async(self.slurm_adapter.disintegrate_machine, drone_uuid="test")
+        )
+
+    def test_drain_machine(self):
+        run_async(self.slurm_adapter.drain_machine, drone_uuid="VM-1")
+        self.mock_async_run_command.assert_called_with(
+            "scontrol update NodeName=host-10-18-1-1 State=DRAIN Reason='COBalD/TARDIS'"
+        )
+
+        self.mock_async_run_command.reset_mock()
+
+        run_async(self.slurm_adapter.drain_machine, drone_uuid="VM-1")
+        self.mock_async_run_command.assert_called_with(
+            "scontrol update NodeName=host-10-18-1-1 State=DRAIN Reason='COBalD/TARDIS'"
+        )
+        self.assertIsNone(
+            run_async(self.slurm_adapter.drain_machine, drone_uuid="not_exists")
+        )
+        self.mock_async_run_command.side_effect = CommandExecutionFailure(
+            message="Does not exists", exit_code=1, stderr="Does not exists"
+        )
+        self.assertIsNone(
+            run_async(self.slurm_adapter.drain_machine, drone_uuid="test")
+        )
+
+        # Review: Not working, and I don't know why. What is this testing?
```

Comment:
This is HTCondor specific. Exit code 1 is returned if HTCondor can't connect to a StartD of a resource. This error is handled in https://github.com/MatterMiners/tardis/blob/cefc49c18bcc4d16910f1858793bc59a8dd17501/tardis/adapters/batchsystems/htcondor.py#L136-L143. If the exit code is different than 1, an Exception should be raised. So, the problem is that no exception at all is raised in your adapter.

Reply:
Thanks, but the comment was referring to `with self.assertRaises(...)`. But your previous comment also made me understand what is going on here :)